code/crates/engine/src/consensus.rs

```diff
@@ -951,12 +954,17 @@ where
                 // NOTE: The message has already been append to the WAL by the `WalAppendMessage` effect.
                 self.wal_flush(phase).await?;
 
-                // Notify any subscribers that we are about to publish a message
-                self.tx_event.send(|| Event::Published(msg.clone()));
+                let should_broadcast = match &msg {
+                    SignedConsensusMsg::Vote(_) => true,
+                    _ => value_payload.include_proposal(),
+                };
```

Comment:
Let's refactor this into a function that we can use here and for the WAL below

Reply:
I think for the WAL we will have to remove the check when we fix #897. So we have three options
- refactor now and undo when we fix #897 
- leave things as they are
- fix #897 in this PR